File: 1684-count-the-number-of-consistent-strings/1684-count-the-number-of-consistent-strings.c
int countConsistentStrings(char * allowed, char ** words, int wordsSize){
    int i,j,k;
    int c = 0;
    
    i = 0;
    while(i < wordsSize)
    {
        j = 0;
        k = 0;
        while(allowed[j])
        {
            if(words[i][k] == NULL)
                break;
            else if(words[i][k] == allowed[j])
            {
                ++k;
                j = -1;
            }
            ++j;
        }
        if(strlen(words[i]) == k)
            c += 1;
        ++i;
    }
    return c;
}

File: 1684-count-the-number-of-consistent-strings/1684-count-the-number-of-consistent-strings.java
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        int i,j,valid = 0;
        HashSet<Character> set = new HashSet<>();
        
        for(i = 0 ; i < allowed.length(); i++)
            set.add(allowed.charAt(i));
        
        for(i = 0; i < words.length; i++){
            for(j = 0; j < words[i].length(); j++){
                if(!set.contains(words[i].charAt(j)))
                    break;
            }
            if(j == words[i].length())
                valid++;
        }
        return valid;
    }
}